src/library_async.js

```diff
@@ -50,6 +50,22 @@ mergeInto(LibraryManager.library, {
     asyncPromiseHandlers: null, // { resolve, reject } pair for when *all* asynchronicity is done
     sleepCallbacks: [], // functions to call every time we sleep
 
+#if ASYNCIFY == 2
+    // TODO: Stack switching support could be implemented without all the
+    //       asyncify infrastructure, perhaps as code on the size instead of
+    //       ifdefs. That might be cleaner, and it would avoid overhead like the
+    //       malloc() asyncify does (we could ifdef it out, but it's even more
+    //       ifdefing).
+
+    // The global suspender object used with the VM's stack switching Promise
+    // API.
+    suspender: null,
+    // The promise that is being suspended on in the VM atm, or null.
+    promise: null,
```

Comment:
What happens if I make a call into the module while the is promise oustanding?

Can we at least assert if I try to do that?

Reply:
I think that should just work. What won't work is trying to suspend again IIUC.

Reply:
I see, thanks! We should probably move over to the static API here eventually, then, even before fibers, since we do want to support "small" calls back into the wasm during a suspend, e.g. to do a malloc. OTOH, maybe that's not urgent as malloc would not be wrapped, so it should just work already.